**Python Assignment 3: While Loop, For loop and Function**

**While loop & Control Statements :Number Guessing Game**

Problem Statement: Create a Python program that implements a simple number 
guessing game using a while loop. The program should make use of control 
statements such as else, break, and continue. 

Instructions: 
1. Set Up the Game: 
* Generate a random number between 1 and 10 that the user has to guess. 
Import random and use randint function.
2. Prompt the User: 
* Ask the user to guess the number. 
* Set a variable attempts to 3, which represents the maximum number of 
guesses allowed. 
3. Implement the Guessing Logic: 
* Use a while loop to allow the user to keep guessing until they get the 
correct number or run out of attempts. 
* Provide feedback to the user for each guess: 
* If the guess is out of the valid range (1 to 10), inform the user. 
* If the guess is greater than the secret number. 
* If the guess is lower than the secret number. 
* If the guess is correct, congratulate the user and end the game. 
4. Control Statements: 
* Use continue to skip the rest of the loop after informing the user if the 
guess is out of range.
* Use break to exit the loop when the guess is correct. 
* Use else with the while loop to provide a message like "Better luck next 
time!" if the user runs out of attempts without guessing the correct number.

In [25]:
import random
secret_number = random.randint(1,10)
attempts=3

while attempts>0:
 guess=int(input("Guess the number(between 1 and 10): "))
      
 if guess<1 or guess>10:
   print("Your guess is out of range, please guess a number between 1 to 10.")
   continue
 if guess==secret_number:
    print("Congrats! Your guess is correct.")
    break
 elif guess>10:
    print("Too high. Try again.")
 else:
    print("Too low. Try again.")
    attempts-=1
         
else:
 print("Better luck next time!")

Guess the number(between 1 and 10):  2


Too low. Try again.


Guess the number(between 1 and 10):  15


Your guess is out of range, please guess a number between 1 to 10.


Guess the number(between 1 and 10):  5


Too low. Try again.


Guess the number(between 1 and 10):  3


Too low. Try again.
Better luck next time!


* It's number guessing game that picks a secret number between 1 and 10, the player have 3 chances to guess it.
* if guess<1 or guess>10: indicates out of range and gives an output message for the player to select between 1 and 10.
* if guess==secret_number: End of game, and congratualing message will displays.
* elif guess>10: indicates the range is high and giving another chance for guessing.
* else: if it's too low from the range and then attempts are over by giving a message with better luck next time.

**For Loop:  Multiplication Table Generator**

Problem Statement: Create a Python program that generates and prints a 
multiplication table (from 1 to 10) for a given number using a for loop and the range 
function. 

Step wise Instructions: 
1. Prompt user for Input. Ask the user to enter a number for which they want to 
generate a multiplication table. 
2. Generate the Multiplication Table: Use a for loop to iterate through the numbers 
1 to 10. In each iteration, calculate the product of the user's number and the 
current number from the loop. 
3. Display the Multiplication Table:  
Print each line of the multiplication table in the format: "number x i = result". 



In [30]:
num=int(input("Enter a number: "))
for i in range(1,11):
  print(f"{num}x{i}= {num*i}") 

Enter a number:  4


4x1= 4
4x2= 8
4x3= 12
4x4= 16
4x5= 20
4x6= 24
4x7= 28
4x8= 32
4x9= 36
4x10= 40


* num=int(input("Enter a number: ")) prompts the user for input. 
* for i in range(1,11): loop then iterates i through 1 to 10
* print(f"{num}x{i}= {num*i}") uses an f-string to display the multiplication expression and its result on each iteration.

**Function: BMI Calculator**

Problem Statement: Create a Python program that calculates the Body Mass Index 
(BMI). 

Hint: BMI = weight (kg) / [height (m)]²  

Instructions: 
1. Define a function calculate_bmi(weight, height) that returns the BMI. 
2. Prompt the user for their weight (in kg) and height (in meters). 
3. Use the function to calculate and display the BMI 

In [2]:
def calculate_bmi(weight,height):
 return(weight/height**2)
weight=float(input("Enter your weight in kg: "))
height=float(input("Enter your height in meters: "))
bmi=calculate_bmi(weight,height)

print((f"Your BMI is: {bmi:.2f}"))

Enter your weight in kg:  58
Enter your height in meters:  1.62


Your BMI is: 22.10


* The code defines calculate_bmi(weight, height) which returns weight / height**2. 
* The prompts for weight in kg and height in meters as floats, calls the function to compute the BMI.
* Prints it formatted to two decimals with an f-string. 
* Results with 58 kg and 1.62 m, it outputs "Your BMI is: 22.10".